framework-support/jobrunr-spring-boot-2-starter/src/main/java/org/jobrunr/scheduling/JobInterceptor.java

```diff
@@ -0,0 +1,53 @@
+package org.jobrunr.scheduling;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.jobrunr.jobs.JobDetails;
+import org.jobrunr.jobs.annotations.Job;
+import org.jobrunr.server.runner.ThreadLocalJobContext;
+import org.jobrunr.utils.JobUtils;
+import org.springframework.core.annotation.AnnotatedElementUtils;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JobInterceptor implements MethodInterceptor {
+    private final JobScheduler jobScheduler;
+
+    public JobInterceptor(JobScheduler jobScheduler) {
+        this.jobScheduler = jobScheduler;
+    }
+
+    @Nullable
+    @Override
+    public Object invoke(@Nonnull MethodInvocation invocation) throws Throwable {
+        Method method = invocation.getMethod();
+        Object[] args = invocation.getArguments();
+
+        // If arguments are present, throw an exception (JobRunr limitation)
+        if (args.length > 0) {
+            throw new IllegalStateException("JobRunr does not support method arguments");
+        }
```

Comment:
For this type of job (a normal `job.enqueue()`, we do accept method arguments. 

It's only for recurring jobs that method arguments don't make sense. So, we can definitely have arguments.